claim/netdata-claim.sh.in

```diff
@@ -85,15 +85,22 @@ ERROR_MESSAGES[17]="Service Unavailable"
 
 # Exit code: 18 - Agent unique id is not generated yet.
 
+NETDATA_RUNNING=1
+
 get_config_value() {
     conf_file="${1}"
     section="${2}"
     key_name="${3}"
-    config_result=$(@sbindir_POST@/netdatacli 2>/dev/null read-config "$conf_file|$section|$key_name"; exit $?)
-    # shellcheck disable=SC2181
-    if [ "$?" != "0" ]; then
-       echo >&2 "cli failed, assume netdata is not running and query the on-disk config"
-       config_result=$(@sbindir_POST@/netdata 2>/dev/null -W get2 "$conf_file" "$section" "$key_name" unknown_default)
+    if [ "${NETDATA_RUNNING}" -eq 1 ]; then
+        config_result=$(@sbindir_POST@/netdatacli 2>/dev/null read-config "$conf_file|$section|$key_name"; exit $?)
+        result="$?"
+        if [ "${result}" -ne 0 ]; then
+            echo >&2 "Unable to communicate with Netdata daemon, querying config from disk instead."
+            NETDATA_RUNNING=0
+        fi
+    fi
+    if [ "${result}" -ne "0" ] || [ "${NETDATA_RUNNING}" -eq 0 ]; then
```

Comment:
I think we only need to check `NETDATA_RUNNING` and not `result`.

Reply:
You’re right, we do only need to check the `NETDATA_RUNNING` variable here.